src/App.tsx

```diff
@@ -1,61 +1,131 @@
+import React, { useState } from 'react';
 import './App.scss';
 
 // import usersFromServer from './api/users';
 // import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number) {
+  const findUser = usersFromServer.find(user => user.id === userId);
+
+  return findUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [titleErrorMessage, setTitleErrorMessage] = useState('');
+  const [userErrorMessage, setUserErrorMessage] = useState('');
+  const [allTodos, setAllTodos] = useState(todos);
+
+  const handleTitleChange: React.ChangeEventHandler<HTMLInputElement>
+    = (event) => {
+      event.preventDefault();
+
+      setTitleErrorMessage('');
+
+      return setTitle(event.target.value);
```

Comment:
```suggestion
      setTitleErrorMessage('');
      setTitle(event.target.value);
```
you don't need prevent default for ChangeEvent. as well as return statement

Reply:
check everywhere